src/main/java/com/github/calebwhiting/runelite/plugins/actionprogress/detect/GuardianOfTheRift.java

```diff
@@ -0,0 +1,144 @@
+
+	private boolean checkInMainRegion(){
+		int[] currentMapRegions = client.getMapRegions();
+		return Arrays.stream(currentMapRegions).anyMatch(x -> x == MINIGAME_MAIN_REGION);
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage chatMessage)
+	{
+		if(!checkInMainRegion()) return;
+		if(chatMessage.getType() != ChatMessageType.SPAM && chatMessage.getType() != ChatMessageType.GAMEMESSAGE) return;
+
+		String msg = chatMessage.getMessage();
+		Matcher rewardPointMatcher = REWARD_POINT_PATTERN.matcher(msg);
+		
+		if(msg.contains("You found some loot:")){
+			currentElementalRewardPoints--;
+			currentCatalyticRewardPoints--;
+		}
+		else if(msg.contains("The Great Guardian was defeated!")){
+			intterupManager.interrupt("Guardian of the Rift game concluded");
+		}
+		else if(rewardPointMatcher.find()) {
+			intterupManager.interrupt("Guardian of the Rift game concluded");
+			currentElementalRewardPoints = Integer.parseInt(rewardPointMatcher.group(1).replaceAll(",", ""));
+			currentCatalyticRewardPoints = Integer.parseInt(rewardPointMatcher.group(2).replaceAll(",", ""));
+		}
+	}
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked evt)
+	{
+		if(!checkInMainRegion()) return;
+
+		if (evt.getId() == REWARD_GUARDIAN_ID && evt.getMenuOption().toLowerCase().equals("search")){
+			triggeringRewardSearch = true;
+		}
+
+		if(evt.getId() == WORKBENCH_ID && evt.getMenuOption().toLowerCase().equals("work-at")){			
+			triggeringEssenceCrafting = true;
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick evt)
+	{
+		if(!checkInMainRegion()) return;
+
+		Widget dialog = client.getWidget(DIALOG_WIDGET_GROUP, DIALOG_WIDGET_MESSAGE);
+		if (dialog != null){			
+		String dialogText = dialog.getText();
+		final Matcher checkMatcher = CHECK_POINT_PATTERN.matcher(dialogText);
+		if (checkMatcher.find(0))
+		{
+			currentCatalyticRewardPoints = Integer.parseInt(checkMatcher.group(1));
+			currentElementalRewardPoints = Integer.parseInt(checkMatcher.group(2));
+			}
+		}
+		
+		Player local = this.client.getLocalPlayer();
+		if (local == null || local.getAnimation() == -1) return;
+
+		log.debug (local.getAnimation() + " local animation");
+		
+		if (triggeringEssenceCrafting && local.getAnimation() == CRAFTING_ANIMATION_ID){
+			triggeringEssenceCrafting = false;
+
+			ItemContainer inventory = client.getItemContainer(InventoryID.INVENTORY);
+			
+			int numberOfGuardianEssencesToMake = inventory == null ? 0 : Math.min(inventory.count(GUARDIAN_FRAGMENTS_ITEM_ID), this.inventoryManager.getFreeSpaces());
+
+			if (numberOfGuardianEssencesToMake == 0) return;
+			
+			this.actionManager.setAction(Action.GUARDIAN_OF_THE_RIFT_CRAFTING, numberOfGuardianEssencesToMake, ItemID.GUARDIAN_ESSENCE);	
+		}
+		else if (triggeringRewardSearch && local.getAnimation() == REWARD_SEARCHING_ANIMATION_ID){
+			triggeringRewardSearch = false;
+
+			int numberOfRewards = Math.min(currentElementalRewardPoints, currentCatalyticRewardPoints);
```

Comment:
I wonder if the comment above applies to this as well. I will try to remember to specifically look for this next time I play GOTR but it is a bit harder to continually test since it takes some time to generate rewards.

Reply:
Ticks for rewards are working perfectly, no changes needed!